lib/glific/providers/maytapi/message.ex

```diff
@@ -61,4 +85,29 @@ defmodule Glific.Providers.Maytapi.Message do
       }
     }
   end
+
+  @spec get_phone_id(map()) :: non_neg_integer()
+  defp get_phone_id(attrs) do
+    WAManagedPhone
+    |> where([g], g.phone == ^attrs.phone)
+    |> select([g], g.phone_id)
+    |> Repo.one!()
+  end
+
+  @spec create_message_after_send(non_neg_integer(), map()) :: any()
+  defp create_message_after_send(org_id, attrs) do
+    message_attrs =
+      %{
+        body: attrs.message,
+        status: "sent",
+        type: "text",
+        receiver_id: Partners.organization_contact_id(org_id),
+        organization_id: org_id,
+        sender_id: Partners.organization_contact_id(org_id),
+        message_type: "WABA+WA",
+        bsp_status: "sent"
+      }
+
+    Messages.create_message(message_attrs)
```

Comment:
Do we need to put this in an Oban job like we do with gupshup messages to make sure sending message retries on failure? @AkhileshNegi

Reply:
yes @AmishaBisht you need to use Oban to create job and then process jobs
Refer this file`lib/glific/communications/message.ex` where handler is chosen to send message  and then refer this file `lib/glific/providers/gupshup/message.ex` where oban jobs  are created
You need to check this file then where jobs are are processed
`lib/glific/providers/gupshup/worker.ex`